src/viam/sdk/module/service.cpp

```diff
@@ -233,21 +233,38 @@ std::shared_ptr<Resource> ModuleService::get_parent_resource_(const Name& name)
     return parent_->resource_by_name(name);
 }
 
-ModuleService::ModuleService(std::string addr)
-    : module_(std::make_unique<Module>(std::move(addr))), server_(std::make_unique<Server>()) {
+ModuleService::ModuleService(std::string addr) : ModuleService(std::move(addr), "unix:") {}
+
+ModuleService::ModuleService(std::string addr, std::string grpc_conn_protocol)
+    : module_(std::make_unique<Module>(std::move(addr))),
+      grpc_conn_protocol_(std::move(grpc_conn_protocol)),
+      server_(std::make_unique<Server>()) {
     impl_ = std::make_unique<ServiceImpl>(*this);
 }
 
+namespace {
```

Comment:
minor style nit: generally, make one unnamed namespace up at the top of the file, after you've opened the file's main namespace:

```
#include <my_header.hpp>

#include <stuff>

namespace my {
namespace thing {

namespace {

// put get_protocol here

}  // namespace

...

}  // namespace thing
}  // namespace my
```

It's a convention, there is no real harm to opening and closing the unnamed namespace as often as you need, but keeping all the little utilities up together in the unnamed namespace up at the top prevents a proliferation of tiny namespace blocks.